app/controllers/jobs_controller.rb

```diff
@@ -21,24 +29,53 @@ def create
   end
 
   def show
-    @job = Job.find(params[:id])
+    @comment = Comment.new
   end
 
   def edit
-    # implement on your own!
   end
 
   def update
-    # implement on your own!
+    if @job.update(job_params)
+      redirect_to company_job_path(@company, @job)
+    else
+      @job.errors.full_messages
+      render :edit
+    end
   end
 
   def destroy
-    # implement on your own!
+    @job.destroy
+    redirect_to company_jobs_path(@company)
+  end
+
+  def by_location
+    @jobs = Job.where(city: params[:location])
+    render :by_location
+  end
+
+  def sorter
+    if params["sort"] == "location"
+      @sorted = "Location"
+      @sort_by = Job.all.group(:city).count("id")
+    else
+      @sorted = "Level of Interest"
+      @sort_by = Job.all.group(:level_of_interest).count("id")
+    end
+    render :sorter
+  end
+
+  def find_job
```

Comment:
since this is just used internally within this class, i suggest making it a private method! same for `find_company`